doc/man3/hpke_enc.pod

```diff
@@ -0,0 +1,241 @@
+called <foo>len) and an unsigned char* pointer <foo>. For outputs buffers,
+the length on input is the caller-allocated buffer size, and the size-used on
+successful output.  Buffers for plaintexts and ciphertexts can be any size, all
+other buffers (e.g. for key material, aad, etc.) can be at most HPKE_MAXSIZE.
+
+- authpub is a buffer for the HPKE public value to be used for AUTH modes 
+
+- authpriv is a buffer for the HPKE private value to be used for AUTH modes
+
+- authpriv_evp in an EVP_PKEY* equivalent of authpriv (either can be supplied,
+if both are, authpriv_evp is preferred)
+
+- pskid is a char* containing a string identifying a PSK
+
+- psk is a buffer containing a pre-shared key
+
+- clear is a buffer for input cleartext to be encrypted or a buffer for 
+recovered plaintext, after decryption
+
+- aad is a buffer for additional authenticated data
+
+- info is a buffer for information to be bound to the HPKE exchange (generally
+to "fold in" identity information)
+
+- senderpub is a buffer for the ECDH public value generated by the 
+sender (encryptor) and to be used by the recipient (decryptor) 
+
+- senderpriv is the EVP_PKEY* form of the private value corresponding to
+senderpub
+
+- pub is the receiver's non-ephemeral ECDH public value for encryption
+
+- priv is the receiver's non-ephemeral ECDH private value for decryption
+
+- seq is a buffer representing the sequencing of encryptions when multiple
+calls are made (to be XOR'd with an internal nonce)
+
+- cipher is a buffer for ciphertext (not including the recipient's public
+value)
+
+hpke_expansion() provides a way for the caller to know by how much cipertext
+is longer than plaintext.
+
+hpke_good4grease() produces values of the appropriate length (for the given
+ciphersuite) so that a protocol using HPKE can send so-called GREASE values
+that are harder to distinguish from a real use of HPKE.
+
+hpke_setlibctx() allows the caller to use a nondefault OSSL_LIB_CTX.
+
+
+=head1 RETURN VALUES
+
+All functions return 1 for success. Functions may return a negative 
+number, zero or a positive number in the event of an error.
+
+=head1 EXAMPLES
+
+This example does a minimal round-trip encryption using HPKE.
+
+
+    #include <crypto/hpke.h>
+    /* we'll do a round-trip, generating a key, encrypting and decrypting */
+    int hpke_mode=HPKE_MODE_BASE;
+    hpke_suite_t hpke_suite = HPKE_SUITE_DEFAULT;
+    /* we'll alloc all these on the stack for simplicity */
+    size_t publen=HPKE_MAXSIZE; unsigned char pub[HPKE_MAXSIZE];
+    size_t privlen=HPKE_MAXSIZE; unsigned char priv[HPKE_MAXSIZE];
+    size_t senderpublen=HPKE_MAXSIZE; unsigned char senderpub[HPKE_MAXSIZE];
+    size_t plainlen=HPKE_MAXSIZE; unsigned char plain[HPKE_MAXSIZE];
+    size_t cipherlen=HPKE_MAXSIZE; unsigned char cipher[HPKE_MAXSIZE];
+    size_t clearlen=HPKE_MAXSIZE; unsigned char clear[HPKE_MAXSIZE];
+    memset(plain,0,HPKE_MAXSIZE);
```

Comment:
nit: blank line after declaration

Reply:
fixed in 32f0f8a64b10ff8c503b04fe71cbe47a7aabb968